lib/cli.rb

```diff
@@ -827,19 +837,36 @@ def self.check_port_validity(port)
       desc: "Listen on port PORT. Ports can be selected in a range between 2-65535. Ports between
         2 and 1023 can only be chosen when `machinery` will be executed as `root` user.",
         arg_name: "PORT"
-    c.flag [:ip, :i], type: String, required: false, default_value: "127.0.0.1",
-      desc: "Listen on ip address IP. It's only possible to use an IP address (or hostnames
-        resolving to an IP address) which is assigned to a network interface on the local
-        machine.", arg_name: "IP"
+    c.switch "public", required: false, negatable: false,
+      desc: "Makes the server reachable from all IP addresses."
 
     c.action do |_global_options, options, args|
       name = shift_arg(args, "NAME")
 
-      check_port_validity(options[:port])
+      begin
+        check_port_validity(options[:port])
+      rescue Machinery::Errors::InvalidServerPortSpecified
+        raise Machinery::Errors::InvalidCommandLine.new("You have to specify a valid server " \
+          "port either in the 'http_server_port' section of the configuration file or " \
+          "via the --port option. The specified port '#{options[:port]}' is not valid. A valid " \
+          "port can be in a range between 2 and 65535.")
+      rescue Machinery::Errors::ServerNeedRootPrivileges
+        raise Machinery::Errors::InvalidCommandLine.new("You specified the port " \
+          "'#{options[:port]}' either in the 'http_server_port' section of the configuration " \
+          "file or via the --port option. To start the server, you need root privileges. If " \
+          "you don't want to start the server as root user, you must choose a port between " \
+          "1024 and 65535.")
```

Comment:
The error messages are duplicated here. There should only be one place where they are defined.